docs/reference/mapping/types/runtime.asciidoc

```diff
@@ -0,0 +1,585 @@
+[[runtime]]
+=== Runtime
+Typically, you must index fields to {es} before they can be retrieved,
+aggregated, or searched. With _runtime fields_ beta[], you can explicitly
+define a field in the mapping and access it at search time without indexing
+your data during ingest time.
+
+Because runtime fields aren't indexed, you can more quickly ingest raw data
+into the Elastic Stack and access it immediately. Runtime fields are accessible
+from the search API like any other field. The <<search-field-caps,field capabilities API>> returns runtime fields as both searchable and aggregatable.
+
+By dynamically evaluating runtime fields at search time, you can optimize disk
+space by choosing which fields to index. If you later decide that a runtime
+field is increasingly used for filtering or aggregation, you can add the field
+to `_source` to gain faster search speed.
+
+Alternatively, you can specify a `runtime_mappings` section in a search request
+to use runtime fields that exist only as part of the query. This
+flexibility allows you to create or override field values in `_source` for the
+duration of the query without modifying the field itself.
+
+Runtime fields help to alleviate several common issues when using {es}:
+
+* Reindexing your data between development iterations is slow and can make
+experimenting on large datasets difficult
+* Indexing data before searching makes running one-off searches costly and
+resource intensive
+* Indexing all of your data instead of just the fields you want to search
+requires more disk space to gain performance at search time
+* Reindexing data for time-based indices to ensure that existing indices
+include any new fields in the index template is slow
+* Determining how a field is evaluated in {kib} at index or search time is
+difficult for scripted fields because they have different needs based on where
+they are implemented
+
+Runtime fields incur performance costs at search time, depending
```

Comment:
Consider discussing the correct choice instead, e.g. if you choose to index fields that you know you will filter, search and aggregate on, like timestamp, then Elasticsearch will automatically use these indexed fields first in the query execution process. The result will be that the system will end up calculating the value of the runtime fields for a relatively small subset of the documents, so although runtime fields are slower than indexed fields, the query performance degradation will be limited. Runtime fields and indexed fields complement each other and work best in tandem.

Reply:
@giladgal, I created a new [Compromises](https://elasticsearch_62653.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/runtime.html#runtime-compromises) section, which hopefully encompasses the sentiments that you're discussing.